modules/cluster-logging-configuration-of-json-log-data-for-default-elasticsearch.adoc

```diff
@@ -0,0 +1,71 @@
+[id="cluster-logging-configuration-of-json-log-data-for-default-elasticsearch_{context}"]
+= Configuration of JSON log data for default Elasticsearch
+
+With the default Elasticsearch log store, you must create an index for each format if the JSON log entries _have different formats_.
+
+[IMPORTANT]
+====
+You must create a separate index for each different JSON log format. Otherwise, forwarding different formats to the same index can cause type conflicts and cardinality problems.
+====
+
+To provide a different index for each format, you configure the `ClusterLogForwarder` custom resource. You use a structure type to construct the index name from a category, Kubernetes label, or input selector.
+
+.Structure types
+
+The `ClusterLogForwarder` provides the following structure types for use with the default Elasticsearch log store:
+
+* `structuredTypeName`: (string, optional) If `structuredTypeKey` is not set or its key is not present, use the value of `structuredTypeName` as the structured type.
+* `structuredTypeKey`: (string, optional) If either of the following meta-data keys is set and its value is non-empty, use the value of the key as the structured type.
+** `kubernetes.labels.<key>`: Use the string value of the `kubernetes` label that has the specified `<key>`.
+** `openshift.labels.<key>`: Use the string value of the `openshift` label that has the specified `<key>`.
```

Comment:
Use the string value of the `openshift` label that specified in the Clusterlogfoward resource.

Reply:
@anpingli Does this mean that, with the following example snippet from `ClusterLogForwarder`, for `<key>` the user can specify `logs`? And OpenShift Logging will generate an index named `application`? If not, @alanconway, would you please explain?

```
  pipelines:
   - name: application-logs
     inputRefs:
     - application
     - audit
     outputRefs:
     - elasticsearch-secure
     - default
     parse: json
     labels:
       logs: application
```

Reply:
structuredTypeKey is the name of a message field. The value of that field (if present) is used to construct the index name.
- kubernetes.labels.foo - use the value of the kubernetes Pod label "foo" to form the index name.
- openshift.labels.foo - use the value of the label "foo" specified in the CLF pipeline "labels" configuration (it allows you to mark messages with static labels based on the CLF input selectors)

Reply:
Updated. Please review.